src/Compilers/Core/Portable/SymbolDisplay/SymbolDisplayParameterOptions.cs

```diff
@@ -63,5 +63,11 @@ public enum SymbolDisplayParameterOptions
         /// Includes square brackets around optional parameters.
         /// </summary>
         IncludeOptionalBrackets = 1 << 5,
+
+        /// <summary>
+        /// Equivalent to <see cref="IncludeName"/> but only if the parameter is displayed on its own
+        /// (i.e., not as part of a method, delegate, or indexer).
+        /// </summary>
+        IncludeNameIfStandalone = 1 << 6,
```

Comment:
>IncludeNameIfStandalone [](http://example.com/codeflow?start=8&length=23)

If this is a change to the public API, we'll want to include in API review.

Reply:
i'm a big confused by this too.  if the parameter is on its own... why not still just use `IncludeName`?  effectively: if `IncludeName` is present... then include the name :D.  The fact that it doesn't seems like the strange thing here.

Reply:
> if `IncludeName` is present... then include the name

This is about changing the default setting to improve error messages. If `IncludeName` was enabled for all parameters, it would result in lots of (test) changes for all method types, etc. In other words, I want to display `Method(int, string)` but `int i`, `string s` in error messages by default.

Reply:
>  In other words, I want to keep displaying Method(int, string) but also display int i, string s in error messages by default.

I feel like that should be a *member* option.  e.g. how the member displays parameters.

Reply:
That makes sense. For now, I changed it to be `internal` to avoid API changes: 81aff40e22288f3ee596f19b8d048a4eb72a1bb0. It should be easy to change the API to whatever, so we can decide to introduce a public API instead in the future.